_examples/golang-basics/example.gen.go

```diff
@@ -78,13 +78,44 @@ func (x *Kind) Is(values ...Kind) bool {
 	return false
 }
 
+type Intent string
+
+const (
+	Intent_openSession     Intent = "openSession"
+	Intent_closeSession    Intent = "closeSession"
+	Intent_validateSession Intent = "validateSession"
+)
+
+func (x Intent) MarshalText() ([]byte, error) {
+	return []byte(x), nil
+}
+
+func (x *Intent) UnmarshalText(b []byte) error {
+	*x = Intent(string(b))
+	return nil
+}
+
+func (x *Intent) Is(values ...Intent) bool {
+	if x == nil {
+		return false
+	}
+	for _, v := range values {
+		if *x == v {
+			return true
+		}
+	}
+	return false
+}
+
 type Empty struct {
 }
 
 type User struct {
-	ID       uint64 `json:"id" db:"id"`
-	Username string `json:"USERNAME" db:"username"`
-	Role     string `json:"role" db:"-"`
+	ID       uint64  `json:"id" db:"id"`
+	Username string  `json:"USERNAME" db:"username"`
+	Role     string  `json:"role" db:"-"`
+	Kind     *Kind   `json:"kind"`
```

Comment:
perhaps its for historical reasons, but do you guys remember why we default to using a `*` for the enum values..? as these are just integers or strings, I feel we should drop the pointer... thoughts..?

Reply:
FYI, if someone defines a field as, `kind?: Kind` then yes it should output *Kind, but if the definition is just `kind: Kind` then I believe it should just be of type `Kind`

Reply:
see also reported in the past https://github.com/webrpc/webrpc/issues/55

Reply:
I don't see a reason for this. Agreed.

Reply:
It's a problem on `gen-golang` template side see: https://github.com/webrpc/gen-golang/pull/53